### PR TITLE
Add support for Oracle CDC in listening mode

### PR DESCRIPTION
**Pull request by CharukaK**
## Purpose
$subject

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Test environment
JDK 1.8.0_211, OS: Mac OSX, Ubuntu

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,261 @@
-#idea files
```
Comment (pcnfernando) [inline .gitignore]: Why are you updating gitignore

File: component/pom.xml [hunk 2]
```diff
@@ -519,14 +520,16 @@
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
                         <Bundle-Name>${project.artifactId}</Bundle-Name>
                         <Export-Package>
-                            io.siddhi.extension.io.cdc.*;version="${project.version}"
+                            io.siddhi.extension.io.cdc.*;version="${project.version}",
```
Comment (pcnfernando) [inline component/pom.xml]: Why are you both private packaging and exporting. Isn't these dependencies being used within your extension

File: component/pom.xml [hunk 3]
```diff
@@ -519,14 +520,16 @@
...
+                            org.apache.kafka.common.*,
+                            org.apache.kafka.clients.*,
                         </Export-Package>
                         <Private-Package>
                             com.github.shyiko.*,
-                            io.debezium.*,
-                            org.apache.kafka.connect*,
-                            org.apache.kafka.common*,
-                            org.apache.kafka.clients*
+                            io.debezium.*,;version="${debezium-embedded.version}"
```
Comment (pcnfernando) [inline component/pom.xml]: You don't need the version here, since your are private packaging

File: component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java [hunk 4]
```diff
@@ -78,19 +78,65 @@
...
                 "'before_'. e.g., specifying 'before_X' results in the key being added before the column named 'X'." +
-
                 "\n\nFor 'polling' mode: Keys are specified as the columns of the table." +
 
+                "#### Preparations required for working with Oracle Databases in listening mode\n" +
+                "\n" +
+                "Using the extension in Windows, Mac OSX and AIX are pretty straight forward inorder to achieve the " +
+                "required behaviour please follow the steps given below\n" +
+                "\n" +
+                " - First download the following jar versions\n" +
```
Comment (pcnfernando) [inline component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java]: If you are private packaging, why we need kafka.connect dependency jars

File: component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java [hunk 5]
```diff
@@ -307,10 +353,20 @@
...
+                                "operation = 'insert', connector.properties='oracle.outserver.name=DBZXOUT,oracle." +
+                                "pdb=ORCLPDB1' @map(type = 'keyvalue'))\n" +
+                                "define stream insertSweetProductionStream (ID int, NAME string, WEIGHT int);\n",
+                        description = "In this example, the CDC source connect to an Oracle database and listens for" +
+                                " insert queries of sweetproduction table"
                 )
         }
 )
-
+// TODO: 10/28/19 say the configs are mandatory for oracle add an example move the properties as
```
Comment (pcnfernando) [inline component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java]: Remove the todos

File: component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java [hunk 6]
```diff
@@ -331,15 +387,15 @@ protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
...
+                                       String[] requestedTransportPropertyNames, ConfigReader configReader,
+                                       SiddhiAppContext siddhiAppContext) {
         //initialize mode
         mode = optionHolder.validateAndGetStaticValue(CDCSourceConstants.MODE, CDCSourceConstants.MODE_LISTENING);
-
         //initialize common mandatory parameters
         String tableName = optionHolder.validateAndGetOption(CDCSourceConstants.TABLE_NAME).getValue();
         String siddhiAppName = siddhiAppContext.getName();
 
+
```
Comment (pcnfernando) [inline component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java]: Don't add unnecessary new lines

File: component/src/main/java/io/siddhi/extension/io/cdc/source/listening/ChangeDataCapture.java [hunk 7]
```diff
@@ -70,7 +70,7 @@ public void setConfig(Map<String, Object> configMap) {
      * @return {@code engine}.
      */
     public EmbeddedEngine getEngine(EmbeddedEngine.CompletionCallback completionCallback) {
-        // Create and return Engine with above set configuration ...
+        // Create and return Engine with above set configuration ..
```
Comment (pcnfernando) [inline component/src/main/java/io/siddhi/extension/io/cdc/source/listening/ChangeDataCapture.java]: Don't do unnecessary changes or formatting to the existing code

File: component/src/main/java/io/siddhi/extension/io/cdc/util/CDCSourceConstants.java [hunk 8]
```diff
@@ -70,6 +71,9 @@
     public static final int DEFAULT_POLLING_INTERVAL_SECONDS = 1;
     public static final String DATASOURCE_NAME = "datasource.name";
     public static final String JNDI_RESOURCE = "jndi.resource";
+    public static final String ORACLE_PDB_PROPERTY_NAME = "database.pdb.name";
+    public static final String ORACLE_OUTSERVER_PROPERTY_NAME = "database.out.server.name";
+
```
Comment (pcnfernando) [inline component/src/main/java/io/siddhi/extension/io/cdc/util/CDCSourceConstants.java]: Remove unnecessary new lines

File: .gitignore (hunk 1 above)
Comment (CharukaK) [reply .gitignore]: Some autogenerated files were missing which should be ignored

File: component/pom.xml (hunk 2 above)
Comment (CharukaK) [reply component/pom.xml]: they are required for a dependency at the moment without this the extension won't work for Oracle CDC

File: component/pom.xml [hunk 9]
```diff
@@ -510,6 +523,30 @@
...
+                <executions>
+                    <execution>
+                        <id>unpack-dependencies</id>
+                        <phase>compile</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <markersDirectory>${project.build.directory}/dependencies/dependency-maven-plugin-markers</markersDirectory>
+<!--                            <markersDirectory>${project.build.directory}/classes/META-INF/services</markersDirectory>-->
```
Comment (pcnfernando) [inline component/pom.xml]: remove the commented

File: component/pom.xml (hunk 2 above)
Comment (CharukaK) [reply component/pom.xml]: The issue was due to kafka connect-runtime.jar having spi services fixed by copying the services into the manifest folder

File: .gitignore (hunk 1 above)
Comment (CharukaK) [reply .gitignore]: reverted back to the original

File: pom.xml [hunk 10]
```diff
@@ -158,6 +158,23 @@
                 <version>${org.testng.version}</version>
                 <scope>test</scope>
             </dependency>
+
+            <!--     Kafka client dependencies       -->
+            <dependency>
+                <groupId>org.apache.kafka</groupId>
+                <artifactId>connect-json</artifactId>
+                <version>2.3.0</version>
```
Comment (pcnfernando) [inline pom.xml]: Don't hard code the versions. Move them to properties

File: component/src/main/java/io/siddhi/extension/io/cdc/util/CDCSourceUtil.java [hunk 11]
```diff
@@ -126,10 +125,46 @@
...
+                    configMap.put(CDCSourceConstants.ORACLE_PDB_PROPERTY_NAME,
+                            connectorPropertiesMap.get(CDCSourceConstants.ORACLE_PDB_PROPERTY_NAME));
+                    configMap.put(CDCSourceConstants.ORACLE_OUTSERVER_PROPERTY_NAME,
+                            connectorPropertiesMap.get(CDCSourceConstants.ORACLE_OUTSERVER_PROPERTY_NAME));
+                    configMap.put(CDCSourceConstants.CONNECTOR_CLASS, CDCSourceConstants.ORACLE_CONNECTOR_CLASS);
+                    break;
                 }
+                default:
+                    throw new WrongConfigurationException("Unsupported schema. Expected schema: mysql or postgresql" +
+                            "or sqlserver por oracle, Found: " + splittedURL[1]);
```
Comment (dnwick) [inline component/src/main/java/io/siddhi/extension/io/cdc/util/CDCSourceUtil.java:166]: typo here "por"
